### ISSUE TITLE
golangci-lint  failed:[error]/bin/tar:xxxx  Cannot open: File exists

### ISSUE THREAD

**Issue by silenceper** [golangci/golangci-lint-action#23]
https://github.com/silenceper/gowatch/pull/18/checks?check_run_id=722754687
ERROR:
```sh
Installed Go in 77ms
Primary analysis cache key is 'golangci-lint.cache-2630-859920e3b25ff717a0784f9487e35cf40a5fdc1f', restore keys are 'golangci-lint.cache-2630- | golangci-lint.cache-'
Requested golangci-lint 'v1.26', using 'v1.26.0', calculation took 152ms
Installing golangci-lint v1.26.0...
Downloading https://github.com/golangci/golangci-lint/releases/download/v1.26.0/golangci-lint-1.26.0-linux-amd64.tar.gz ...
/bin/tar xz --warning=no-unknown-keyword -C /home/runner -f /home/runner/work/_temp/896dbb3b-38ba-4b3a-80ae-948886b0a1ea
Cache Size: ~11 MB (11729781 B)
/bin/tar -xz -f /home/runner/work/_temp/c9eac72e-6931-4b7d-9a02-499333b10684/cache.tgz -P -C /home/runner/work/gowatch/gowatch
/bin/tar: ../../../go/pkg/mod/github.com/howeyc/fsnotify@v0.9.0/CONTRIBUTING.md: Cannot open: File exists
##[error]/bin/tar: ../../../go/pkg/mod/github.com/howeyc/fsnotify@v0.9.0/fsnotify_test.go: Cannot open: File exists
##[error]/bin/tar: ../../../go/pkg/mod/github.com/howeyc/fsnotify@v0.9.0/example_test.go: Cannot open: File exists
/bin/tar: ../../../go/pkg/mod/github.com/howeyc/fsnotify@v0.9.0/.gitignore: Cannot open: File exists
##[error]/bin/tar: ../../../go/pkg/mod/github.com/howeyc/fsnotify@v0.9.0/fsnotify_open_darwin.go: Cannot open: File exists
/bin/tar: ../../../go/pkg/mod/github.com/howeyc/fsnotify@v0.9.0/CHANGELOG.md: Cannot open: File exists
/bin/tar: ../../../go/pkg/mod/github.com/howeyc/fsnotify@v0.9.0/LICENSE: Cannot open: File exists
```

actions config:
```
name: Go
on: [push,pull_request]
jobs:

  build:
    name: Build
    runs-on: ubuntu-latest
    steps:

      - name: Set up Go 1.x
        uses: actions/setup-go@v2
        with:
          go-version: ^1.13
        id: go

      - name: Check out code into the Go module directory
        uses: actions/checkout@v2

      - name: Get dependencies
        run: |
          go get -v -t -d ./...
          if [ -f Gopkg.toml ]; then
              curl https://raw.githubusercontent.com/golang/dep/master/install.sh | sh
              dep ensure
          fi
      - name: Lint Go Code
        run: |
          export PATH=$PATH:$(go env GOPATH)/bin # temporary fix. See https://github.com/actions/setup-go/issues/14
          go get -u golang.org/x/lint/golint
          go vet ./...
          golint -set_exit_status $(go list ./...)
      - name: Build
        run: go build -v .
```

It is no problem to run the `golangci-lint run` command locally, is my github action configuration wrong?

**Comment by Bluskript** [comment]
I have the same issue

**Comment by jirfag** [comment]
Hi,

it's because you run `go get` before running this action: fetched `$GOPATH/pkg/mod` conflict.

We need to document it better. You can run the action in a separate job, as we recommend in README to fix it.

**Comment by h-michael** [comment]
@jirfag Thanks for developing great apps :)
I got the same issue. But I can't do your suggestion.
Because I use this action for private repository applications which depends on private modules.
So I should run `go get -t -v ./...` with machine user GitHub token before running golangci-lint.
Do I have a solution with this?

**Comment by silenceper** [comment]
> Hi,
> 
> it's because you run `go get` before running this action: fetched `$GOPATH/pkg/mod` conflict.
> 
> We need to document it better. You can run the action in a separate job, as we recommend in README to fix it.

fixed

**Comment by SVilgelm** [comment]
@silenceper Hi, could you provide more details about how you fixed it? I think it can be useful for others

**Comment by mojtabacazi** [comment]
@jirfag @silenceper  I just ran into the same issue. I have to run `go get` to generate mocks and other things required for linter to work. Can you guys explain how it would work?

**Comment by silenceper** [comment]
> @silenceper Hi, could you provide more details about how you fixed it? I think it can be useful for others

I remove "Get dependencies" and is works

**Comment by natefinch** [comment]
Just a note, this seems to happen when I run golangci-lint more than once in the same action. In my case, I have a lenient config I run over the whole repo and then a more strict config I run over the files that have changed in the PR. Setting `skip-pkg-cache: true` on each run of the golangci-lint action seems to fix it. 

I'm still unclear why this bug was closed, though. Running golangci-lint in a separate job is not always possible. In my case, I wrote an action to share with many different go repos, so everything has to be done in the same job.
